### ISSUE TITLE
Construct and compare with random network

### ISSUE THREAD

**Issue by Shawnae-Wang** [ChiLiubio/microeco#216]
Hi. Is there any method to construct classic random network or generalized random graph in this package? And could we further compare our network with the constructed random network to characterize our network as presenting non-random interactions of microbiome? Looking forward to your reply.

**Comment by ChiLiubio** [comment]
Do you mean the random network like this (https://rpubs.com/lgadar/generate-graphs) or constructed based on the community with a null model or from a paper? I have not created a functin to directly generate it, but it is relatively easy to do. I will show an example once I am sure the random network you want to do. Comparing networks can be performed with meconetcomp package (https://github.com/ChiLiubio/meconetcomp).

**Comment by Shawane-Wang** [comment]
I mean constructing a random network using parameters from our empirical network (like count of nodes, edges and so on). Here is an example that might be vivid (https://mp.weixin.qq.com/s?__biz=MzIxNzc1Mzk3NQ==&mid=2247484753&idx=1&sn=8b00273dfee2478478d43fbbc8da7bf3&chksm=97f5b549a0823c5f673d77006ace7641cbb93539537483e6fb15243b37187498573482f3b787&token=1817505529&lang=zh_CN&scene=21#wechat_redirect).

**Comment by ChiLiubio** [comment]
Thanks. I updated the package in github and add a function `random_network`. Please reinstall it from github and have a try.
Here is my test.
```
library(microeco)
data(dataset)
t1 <- trans_network$new(dataset = dataset, cor_method = "bray", filter_thres = 0.003)
t1$cal_network(COR_p_thres = 0, COR_cut = 0.3)
test <- t1$random_network(runs = 100)
test
```

**Comment by Shawane-Wang** [comment]
Thanks! Could we further get results like clustering coefficient, average path length rand and other network parameters according to every run of the random network, which may help us to generate the histograms of these parameters comparing with those from our empirical network? Maybe pictures as following?
![image](https://user-images.githubusercontent.com/75312600/230430647-3d164ea4-3e32-4f0f-9a25-b1f6ca8abac0.png)

**Comment by ChiLiubio** [comment]
Yes. I forgot to tell you the parameter `output_sim` can control whether output each result of simulation. When `output_sim = TRUE`, the columns from five to the last are the simulations.
```
library(microeco)
data(dataset)
t1 <- trans_network$new(dataset = dataset, cor_method = "bray", filter_thres = 0.003)
t1$cal_network(COR_p_thres = 0, COR_cut = 0.3)
test <- t1$random_network(runs = 100, output_sim = TRUE)
test
```
Then you can extract this part of results for the customized visualization.

**Comment by Shawane-Wang** [comment]
I find it! Many thanks!

**Comment by Shawane-Wang** [comment]
In my own dataset, I called t1 as tmp, but it returns as follows. Is it possible to be more flexible when defining our object?
![image](https://user-images.githubusercontent.com/75312600/230532024-3a2bb396-7bc6-48f8-b797-8663538f12bf.png)

**Comment by Shawane-Wang** [comment]
I guess the error may comes from here.
![image](https://user-images.githubusercontent.com/75312600/230532535-f79da3c4-ec37-4c59-b00a-2ffb2e8b8d33.png)

**Comment by Shawane-Wang** [comment]
In my own dataset, I called t1 as tmp, but it returns as follows. Is it possible to be more flexible when defining our object?

![image](https://user-images.githubusercontent.com/75312600/230533567-f8c78378-bd0b-4f99-ab84-dba4c12da299.png)

I guess the error may comes from here.
![image](https://user-images.githubusercontent.com/75312600/230533589-1b4b17b6-6b7a-48f6-89b9-dcb8cf0c7ecc.png)

Besides, would it be possible to add these results to the object? Like the message implied like this “Bootstrap result is stored in object$res_powerlaw_p”? Caz we might have generated many sub-network across group in the first step at the same time and wanted to run loop to plot certain results in the following steps.

**Comment by ChiLiubio** [comment]
Sorry. It is a bug. I have fixed it. Please reinstall the package from github and try again.
Thanks.

**Comment by Shawane-Wang** [comment]
It works! Thank you for your kind help!
